x-pack/plugins/case/server/scripts/sub_cases/index.ts

```diff
@@ -0,0 +1,217 @@
+ * 2.0.
+ */
+/* eslint-disable no-console */
+import yargs from 'yargs';
+import { KbnClient, ToolingLog } from '@kbn/dev-utils';
+import {
+  CaseResponse,
+  CaseType,
+  CollectionWithSubCaseResponse,
+  ConnectorTypes,
+} from '../../../common/api';
+import { CommentType } from '../../../common/api/cases/comment';
+import { CASES_URL } from '../../../common/constants';
+import { ActionResult, ActionTypeExecutorResult } from '../../../../actions/common';
+
+main();
+
+function createClient(argv: any): KbnClient {
+  return new KbnClient({
+    log: new ToolingLog({
+      level: 'info',
+      writeTo: process.stdout,
+    }),
+    url: argv.kibana,
+  });
+}
+
+async function handleFind(argv: any) {
+  const client = createClient(argv);
+
+  try {
+    const res = await client.request({
+      path: `${CASES_URL}/${argv.caseID}/sub_cases/_find`,
+      method: 'GET',
+      query: {
+        status: argv.status,
+      },
+    });
+    console.log(JSON.stringify(res.data, null, 2));
+  } catch (e) {
+    console.error(e);
+    throw e;
+  }
+}
+
+async function handleDelete(argv: any) {
+  const client = createClient(argv);
+
+  try {
+    await client.request({
+      path: `${CASES_URL}?ids=["${argv.id}"]`,
+      method: 'DELETE',
+      query: {
+        force: true,
+      },
+    });
+  } catch (e) {
+    console.error(e);
+    throw e;
+  }
+}
+
+async function handleGenGroupAlerts(argv: any) {
+  const client = createClient(argv);
+
+  try {
+    const createdAction = await client.request<ActionResult>({
+      path: '/api/actions/action',
+      method: 'POST',
+      body: {
+        name: 'A case connector',
+        actionTypeId: '.case',
+        config: {},
+      },
+    });
+
+    let caseID: string | undefined = argv.caseID as string | undefined;
+
+    if (!caseID) {
+      console.log('Creating new case');
+      const newCase = await client.request<CaseResponse>({
+        path: CASES_URL,
+        method: 'POST',
+        body: {
+          description: 'This is a brand new case from generator script',
+          type: CaseType.collection,
+          title: 'Super Bad Security Issue',
+          tags: ['defacement'],
+          connector: {
+            id: 'none',
+            name: 'none',
+            type: ConnectorTypes.none,
+            fields: null,
+          },
+          settings: {
+            syncAlerts: true,
+          },
+        },
+      });
+      caseID = newCase.data.id;
+    }
+
+    console.log('Case id: ', caseID);
+    const executeResp = await client.request<
+      ActionTypeExecutorResult<CollectionWithSubCaseResponse>
+    >({
+      path: `/api/actions/action/${createdAction.data.id}/_execute`,
+      method: 'POST',
+      body: {
+        params: {
+          subAction: 'addComment',
+          subActionParams: {
+            caseId: caseID,
+            comment: {
+              type: CommentType.generatedAlert,
```

Comment:
Should be `GeneratedAlertRequestTypeField = 'generated_alert_request'`

Reply:
Oops, thank you will fix.

Reply:
As we discuss `generated_alert_request` is not needed any more so the code is correct as it is.